Basic-Car-Maintenance/Shared/Onboarding/Views/WelcomeViewAddVehicle.swift

```diff
@@ -9,99 +9,177 @@ import SwiftUI
+            .foregroundStyle(.gray)
+            .padding(.horizontal, 25)
+    }
+    
+    private var addVehicleButton: some View {
+        Button {
+            addVehicle()
+        } label: {
+            Text("Add Vehicle")
+                .fontWeight(.bold)
+                .foregroundStyle(.white)
+                .frame(maxWidth: .infinity)
+                .padding(.vertical, 14)
+                .background(Color("basicGreen").gradient, in: .rect(cornerRadius: 12))
+                .contentShape(.rect)
+        }
+        .padding(15)
+        .padding(.horizontal, 15)
+        .alert(
+            Text("Congratulations! 🎉 🚙"),
+            isPresented: $onboardingViewCompletedAlert
+        ) {
+            Button("OK", role: .cancel, action: {
+                isFirstTime = false
+            })
+        } message: {
+            Text(
+                "From now on, next screen will be the main app screen.",
+                comment: "Shows popup when completed Onboarding tutorial"
+            )
+        }
+    }
+    
+    func addVehicle() {
+        if vehicleName.isEmpty {
+            validationAlertName = true
+        } else if vehicleMake.isEmpty {
+            validationAlertMake = true
+        } else if vehicleModel.isEmpty {
+            validationAlertModel = true
+        } else {
+            UserDefaults.standard.set(vehicleName, forKey: "vehicleName")
+            UserDefaults.standard.set(vehicleMake, forKey: "vehicleMake")
+            UserDefaults.standard.set(vehicleModel, forKey: "vehicleModel")
             
+//            isFirstTime = false
+            onboardingViewCompletedAlert = true
+        }
+    }
+    
+    private var backButton: some ToolbarContent {
+        ToolbarItem(placement: .topBarLeading) {
             Button {
-//                isFirstTime = false
+                dismiss()
             } label: {
-                Text("Welcome 🥳")
-                    .fontWeight(.bold)
-                    .foregroundStyle(.white)
-                    .frame(maxWidth: .infinity)
-                    .padding(.vertical, 14)
-                    .background(Color("basicGreen").gradient, in: .rect(cornerRadius: 12))
-                    .contentShape(.rect)
+                HStack {
+                    Image(systemName: "arrow.left.circle")
+                    Text("Back")
+                }
+                .tint(Color("basicGreen"))
             }
-            .padding(15)
-            .padding(.horizontal, 15)
-        }
-        .frame(maxWidth: .infinity, maxHeight: .infinity)
-        .background {
-            Color(UIColor.secondarySystemBackground)
-                .ignoresSafeArea()
         }
-        .navigationBarBackButtonHidden(true)
-        .toolbar {
-            ToolbarItem(placement: .topBarLeading) {
-                Button {
-                    dismiss()
-                } label: {
+    }
+}
+
+struct TextFieldClearButton: ViewModifier {
```

Comment:
The clear button doesn't work, just remove it for now

Reply:
Are you sure? When I am running simulator, it works:


https://github.com/user-attachments/assets/6b05cad6-a286-4b7b-8448-43415721b1fa

Reply:
Sent in PR as it was, because it is working as presented on attached earlier video.